avm/res/app/session-pool/main.bicep

```
@@ -0,0 +1,272 @@
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource sessionPool 'Microsoft.App/sessionPools@2024-10-02-preview' = {
+  name: name
+  location: location
+  identity: identity
+  properties: {
+    containerType: containerType
+    environmentId: environmentId
+    customContainerTemplate: containerType == 'CustomContainer'
+      ? {
+          containers: containers
+          ingress: {
+            targetPort: targetIngressPort
+          }
+          registryCredentials: registryCredentials
+        }
+      : null
+    dynamicPoolConfiguration: {
+      cooldownPeriodInSeconds: cooldownPeriodInSeconds
+      executionType: 'Timed'
+    }
+    managedIdentitySettings: managedIdentitySettings
+    scaleConfiguration: {
+      maxConcurrentSessions: maxConcurrentSessions
+      readySessionInstances: readySessionInstances
+    }
+    sessionNetworkConfiguration: {
+      status: sessionNetworkStatus
+    }
+    poolManagementType: poolManagementType
+  }
+  tags: tags
+}
+
+resource sessionPool_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock ?? {}) && lock.?kind != 'None') {
+  name: lock.?name ?? 'lock-${name}'
+  properties: {
+    level: lock.?kind ?? ''
+    notes: lock.?kind == 'CanNotDelete'
+      ? 'Cannot delete resource or child resources.'
+      : 'Cannot delete or modify the resource or child resources.'
+  }
+  scope: sessionPool
+}
+
+resource sessionPool_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-04-01' = [
+  for (roleAssignment, index) in (formattedRoleAssignments ?? []): {
+    name: roleAssignment.?name ?? guid(sessionPool.id, roleAssignment.principalId, roleAssignment.roleDefinitionId)
+    properties: {
+      roleDefinitionId: roleAssignment.roleDefinitionId
+      principalId: roleAssignment.principalId
+      description: roleAssignment.?description
+      principalType: roleAssignment.?principalType
+      condition: roleAssignment.?condition
+      conditionVersion: !empty(roleAssignment.?condition) ? (roleAssignment.?conditionVersion ?? '2.0') : null // Must only be set if condtion is set
+      delegatedManagedIdentityResourceId: roleAssignment.?delegatedManagedIdentityResourceId
+    }
+    scope: sessionPool
+  }
+]
+
+@description('The name of the session pool.')
+output name string = sessionPool.name
+@description('The resource ID of the deployed session pool.')
+output resourceId string = sessionPool.id
+
+@description('The name of the resource group in which the session pool was created.')
+output resourceGroupName string = resourceGroup().name
+@description('The management endpoint of the session pool.')
+output managementEndpoint string = sessionPool.properties.poolManagementEndpoint
+
+@description('The principal ID of the system assigned identity.')
+output systemAssignedMIPrincipalId string = sessionPool.?identity.?principalId ?? ''
+
+@description('Optional. Custom container definition.')
```

UDT should be exported, ref spec https://azure.github.io/Azure-Verified-Modules/spec/BCPNFR20/

```suggestion
@export()
@description('Optional. Custom container definition.')
```